btcec/nfield.go

```diff
@@ -0,0 +1,1769 @@
+// Copyright (c) 2013-2014 The btcsuite developers
+// Copyright (c) 2013-2014 Dave Collins
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package btcec
+
+// References:
+//   [HAC]: Handbook of Applied Cryptography Menezes, van Oorschot, Vanstone.
+//     http://cacr.uwaterloo.ca/hac/
+
+// All elliptic curve operations for secp256k1 are done in a finite field
+// characterized by a 256-bit prime.  Given this precision is larger than the
```

Comment:
These comments need to be updated to reflect reality.  This is implementing modular arithmetic on the order of the curve as opposed to the prime.